pkg/services/ngalert/schedule/alert_rule.go

```diff
@@ -86,7 +86,7 @@ func newRuleFactory(
 		}
 		return newAlertRule(
 			ctx,
-			rule.GetKey(),
+			rule.GetKeyWithGroup(),
```

Comment:
This change is needed because when the alert routine is stopped, we call `DeleteStateByRuleUID` and it needs the key with the group.